src/Microsoft.Health.Dicom.Functions/Export/ExportDurableFunction.Activity.cs

```diff
@@ -0,0 +1,96 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.Export;
+using Microsoft.Health.Dicom.Core.Models;
+using Microsoft.Health.Dicom.Core.Models.Export;
+using Microsoft.Health.Dicom.Functions.Export.Models;
+using Microsoft.Health.Dicom.Functions.Extensions;
+using Microsoft.Health.Operations.Functions.DurableTask;
+
+namespace Microsoft.Health.Dicom.Functions.Export;
+
+public partial class ExportDurableFunction
+{
+    /// <summary>
+    /// Asynchronously exports a batch of DICOM files to a user-specified sink.
+    /// </summary>
+    /// <param name="context">The context for the activity.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the <see cref="ExportBatchAsync"/> operation.
+    /// The value of its <see cref="Task{TResult}.Result"/> property contains the number a summary of the export
+    /// operation's progress.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="context"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    [FunctionName(nameof(ExportBatchAsync))]
+    public async Task<ExportProgress> ExportBatchAsync([ActivityTrigger] IDurableActivityContext context, ILogger logger)
+    {
+        EnsureArg.IsNotNull(context, nameof(context));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        ExportBatchArguments args = context.GetInput<ExportBatchArguments>();
+        await using IExportSource source = _sourceFactory.CreateSource(args.Source);
+        await using IExportSink sink = _sinkFactory.CreateSink(args.Destination, context.GetOperationId());
+
+        // Export
+        source.ReadFailure += (source, e) => logger.LogError(e.Exception, "Cannot read desired DICOM file(s)");
```

Comment:
Are these failures for internal tracking or write into the log file shown to customer?

Reply:
Both. The event and logging is for internal use, but the sink should also write the error to the error file